core/tests/webdriverio_utils/ExplorationEditorHistoryTab.js

```diff
@@ -0,0 +1,269 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the exploration editor's history tab, for
+ * use in Webdriverio tests.
+ */
+
+var forms = require('./forms.js');
+var action = require('./action.js');
+var waitFor = require('./waitFor.js');
+
+var ExplorationEditorHistoryTab = function() {
+  /*
+   * Interactive elements
+   */
+  var historyGraph = $('.e2e-test-history-graph');
+  var codeMirrorElementSelector = function() {
+    return $$('.CodeMirror-code');
+  };
+  var toastSuccessElement = $('.toast-success');
+  var firstVersionDropdown = $('.e2e-test-history-version-dropdown-first');
+  var secondVersionDropdown = $('.e2e-test-history-version-dropdown-second');
+  var stateNodeBackground = function(nodeElement) {
+    return nodeElement.$('.e2e-test-node-background');
+  };
+  var stateNodeLabel = function(nodeElement) {
+    return nodeElement.$('.e2e-test-node-label');
+  };
+
+  /*
+   * Buttons
+   */
+  var closeStateHistoryButton = $('.e2e-test-close-history-state-modal');
+  var revertVersionButton = $('.e2e-test-revert-version');
+  var resetGraphButton = $('.e2e-test-reset-graph');
+  var historyListOptionsSelector = function() {
+    return $$('.e2e-test-history-list-options');
+  };
+  var confirmRevertVersionButton = $('.e2e-test-confirm-revert');
+
+  /*
+   * Display
+   */
+  var datesCommitsWereSavedSelector = function() {
+    return $$('.e2e-test-history-tab-commit-date');
+  };
+  /*
+   * Workflows
+   */
+
+  /*
+   * This method checks if the commit dates are being displayed in
+   * the "List of Changes" section of the history tab.
+  */
+  this.expectCommitDatesToBeDisplayed = async function() {
+    var datesCommitsWereSaved = await datesCommitsWereSavedSelector();
+    var numCommitDates = datesCommitsWereSaved.length;
```

Comment:
Basically we only need to worry about this at that places where it takes time to load all the elements in the DOM, like while navigating from chapter editor to story editor it takes some time to load all the chapter list, that why we have waitFor.visibiltyOf before accessing those elements. So its fine to use it like this only if there is no code for visibilityOf present at any place, which basically means we do not need to wait for elements to show up in DOM when we will reach at that point in test suite, it will already be present.